components/raftstore/src/store/read_queue.rs

```diff
@@ -230,14 +216,47 @@ where
         None
     }
 
-    pub fn advance_leader_reads<T>(&mut self, states: T)
+    pub fn advance_leader_reads<T>(&mut self, tag: &str, states: T)
     where
         T: IntoIterator<Item = (Uuid, Option<LockInfo>, u64)>,
     {
-        for (uuid, _, index) in states {
-            assert_eq!(uuid, self.reads[self.ready_cnt].id);
-            self.reads[self.ready_cnt].read_index = Some(index);
-            self.ready_cnt += 1;
+        let mut states_iter = states.into_iter();
+        while let Some((uuid, info, index)) = states_iter.next() {
```

Comment:
with the message filter above, is this block still necessary? or it's just for potential debug purpose?

Reply:
Yes, it's a defense.